server/sublist.go

```diff
@@ -1061,11 +1061,21 @@ func IsValidSubject(subject string) bool {
 	sfwc := false
 	tokens := strings.Split(subject, tsep)
 	for _, t := range tokens {
-		if len(t) == 0 || sfwc {
+		length := len(t)
+		if length == 0 || sfwc {
 			return false
 		}
-		if len(t) > 1 {
-			if strings.ContainsAny(t, "\t\n\f\r ") {
+		if length > 1 {
+			if strings.ContainsAny(t, "\t\n\f\r") {
+				return false
+			}
+			// if the token is a moustache don't worry about spaces inside
+			if length > 4 {
+				if t[0] == '{' && t[1] == '{' && t[length-2] == '}' && t[length-1] == '}' {
```

Comment:
Not really my area of expertise but I am not sure this is the right fix, IsValidSubject is called all over the place to validate subjects not just mappings.  Spaces in subjects are always invalid and IsValidSubject should not say they are.

Where we test subjects in the process of establishing mappings we should handle the specific case where mappings are not valid subjects.

Reply:
Maybe I misunderstood the issue, so you would rather have a new 'isValidMapping' that accepts spaces inside mustaches to be called in https://github.com/nats-io/nats-server/blob/a52f12613e6eb53097d6d563d9b9681b88a037cb/server/accounts.go#L618-L621 ?

Reply:
yes, or just remove the call to IsValidSubject() there, a deep validate might be good though, one that checks for like invalid mapping functions and all that stuff.

Reply:
For sure deeper validate and better errors to let the user know what the problem is a good thing as it's rather poor right now, will expand the scope from just #3151

Reply:
Yeah, I think we can also tackle those in a follow up and just drop the valid subject check here tbh

Reply:
Yeah, that was a bit my idea (better validation/error handling in a follow up), but regardless of the breaking it down or not fair enough on a new IsValidMappingDestSubject() anyways which can be made as simple as what I have done in that initial commit or mode much more sophisticated with better errors.

Reply:
Now does deeper validation, and also returns new kinds of errors to help the users when creating mappings, should be good now.